Build fails on latest Crystal nightly image

`docker run --rm crystallang/crystal:nightly-alpine crystal eval 'puts "foo"'`

Produces:

```sh
/usr/lib/gcc/x86_64-alpine-linux-musl/9.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/bin/../lib/crystal/libgc.a(pthread_support.o): in function `GC_thr_init':
pthread_support.c:(.text+0xe39): undefined reference to `gnu_get_libc_version'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/bin/../lib/crystal/libgc.a(mach_dep.o): in function `GC_with_callee_saves_pushed':
mach_dep.c:(.text+0x3b): undefined reference to `getcontext'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/bin/../lib/crystal/libgc.a(pthread_start.o): in function `GC_inner_start_routine':
pthread_start.c:(.text+0x44): undefined reference to `__pthread_register_cancel'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: pthread_start.c:(.text+0x6b): undefined reference to `__pthread_unregister_cancel'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o /root/.cache/crystal/crystal-run-eval.tmp  -rdynamic -L/usr/bin/../lib/crystal -lpcre -lm -lgc -lpthread -levent -lrt`
```

First noticed via https://github.com/Blacksmoke16/oq/runs/3875659755?check_suite_focus=true.

--- a/docker/alpine.Dockerfile
+++ b/docker/alpine.Dockerfile
@@ -34,6 +34,7 @@ COPY ${crystal_targz} /tmp/crystal.tar.gz
 RUN \
   tar -xz -C /usr --strip-component=1  -f /tmp/crystal.tar.gz \
     --exclude */lib/crystal/lib \
+    --exclude */lib/crystal/*.a \
     --exclude */share/crystal/src/llvm/ext/llvm_ext.o && \
   rm /tmp/crystal.tar.gz
 

